waku/common/rate_limiter.go

```diff
@@ -138,11 +137,20 @@ func NewPeerRateLimiter(cfg *PeerRateLimiterConfig, handlers ...RateLimiterHandl
 }
 
 func (r *PeerRateLimiter) Decorate(p RateLimiterPeer, rw p2p.MsgReadWriter, runLoop runLoop) error {
-	in, out := p2p.MsgPipe()
-	defer in.Close()
-	defer out.Close()
 	errC := make(chan error, 1)
 
+	in, out := p2p.MsgPipe()
+	defer func() {
+		if err := in.Close(); err != nil {
+			errC <- err
+		}
+	}()
+	defer func() {
+		if err := out.Close(); err != nil {
+			errC <- err
+		}
+	}()
+
```

Comment:
[Error handling is good](https://github.com/status-im/status-go/pull/1957/files#r417328424)